packages/m2m-gateway/src/services/purposeService.ts

```diff
@@ -1,8 +1,109 @@
+import { m2mGatewayApi, purposeApi } from "pagopa-interop-api-clients";
+import { WithLogger } from "pagopa-interop-commons";
+import { PurposeId } from "pagopa-interop-models";
+import {
+  toGetPurposesApiQueryParams,
+  toM2MGatewayApiPurpose,
+} from "../api/purposeApiConverter.js";
 import { PagoPAInteropBeClients } from "../clients/clientsProvider.js";
+import { M2MGatewayAppContext } from "../utils/context.js";
+import { WithMaybeMetadata } from "../clients/zodiosWithMetadataPatch.js";
+import {
+  pollResource,
+  isPolledVersionAtLeastResponseVersion,
+} from "../utils/polling.js";
 
 export type PurposeService = ReturnType<typeof purposeServiceBuilder>;
 
 // eslint-disable-next-line @typescript-eslint/explicit-function-return-type
-export function purposeServiceBuilder(_clients: PagoPAInteropBeClients) {
-  return {};
+export function purposeServiceBuilder(clients: PagoPAInteropBeClients) {
+  // eslint-disable-next-line @typescript-eslint/explicit-function-return-type
+  const pollPurpose = (
+    response: WithMaybeMetadata<purposeApi.Purpose>,
+    headers: M2MGatewayAppContext["headers"]
+  ) =>
+    pollResource(() =>
+      clients.purposeProcessClient.getPurpose({
+        params: { id: response.data.id },
+        headers,
+      })
+    )({
+      checkFn: isPolledVersionAtLeastResponseVersion(response),
+    });
+
+  return {
+    async getPurposes(
+      queryParams: m2mGatewayApi.GetPurposesQueryParams,
+      { logger, headers }: WithLogger<M2MGatewayAppContext>
+    ): Promise<m2mGatewayApi.Purposes> {
+      const { eserviceIds, limit, offset } = queryParams;
+
+      logger.info(
+        `Retrieving purposes for eServiceIds ${eserviceIds} limit ${limit} offset ${offset}`
+      );
+
+      const queries = toGetPurposesApiQueryParams(queryParams);
+
+      const {
+        data: { results, totalCount },
+      } = await clients.purposeProcessClient.getPurposes({ queries, headers });
+
+      return {
+        results: results.map((purpose) =>
+          toM2MGatewayApiPurpose({
+            purpose,
+            logger,
+            mapThrownErrorsToNotFound: true,
```

Comment:
True, that's a case I missed in the review of the corresponding PR! 
However, filtering out results at this point in the code breaks the pagination, because then offset/limit/totalcount would not be true anymore 😢  Maybe we should add a filter in the getPurposes of the purpose-process to include only purposes with active versions? And then, here we can remove the `mapThrownErrorsToNotFound: true` and leave the function to throw a 500, because it would be an unexpected case.
@galales opinions?

```suggestion
```